test/blocks/global-footer/test-utilities.js

```diff
@@ -0,0 +1,125 @@
+import { setViewport } from '@web/test-runner-commands';
+import { setConfig } from '../../../libs/utils/utils.js';
+import { config, isElementVisible, loadStyles, viewports } from '../global-navigation/test-utilities.js';
+import { waitForElement } from '../../helpers/waitfor.js';
+
+export const containerSelector = '.global-footer';
+
+export const baseSelectors = {
+  footerIcons: '.feds-footer-icons',
+  footerWrapper: '.feds-footer-wrapper',
+};
+
+export const menuSelectors = {
+  menuContent: '.feds-menu-content',
+  menuColumn: '.feds-menu-column',
+  menuSection: '.feds-menu-section',
+  menuHeadline: '.feds-menu-headline',
+  menuItems: '.feds-menu-items',
+};
+
+export const featuredProductsSelectors = {
+  featuredProducts: '.feds-featuredProducts',
+  featuredProductsLabel: '.feds-featuredProducts-label',
+  navLink: '.feds-navLink',
+  navLinkImage: '.feds-navLink-image',
+  navLinkContent: '.feds-navLink-content',
+  navLinkTitle: '.feds-navLink-title',
+};
+
+export const socialLinksSelectors = {
+  social: '.feds-social',
+  socialItem: '.feds-social-item',
+};
+
+export const regionPickerSelectors = { regionPickerWrapper: '.feds-regionPicker-wrapper' };
+
+export const legalSelectors = {
+  legalWrapper: '.feds-footer-legalWrapper',
+  privacySection: '.feds-footer-privacySection',
+  copyright: '.feds-footer-copyright',
+  privacyLink: '.feds-footer-privacyLink',
+};
+
+export const selectors = {
+  ...baseSelectors,
+  ...menuSelectors,
+  ...featuredProductsSelectors,
+  ...socialLinksSelectors,
+  ...regionPickerSelectors,
+  ...legalSelectors,
+};
+
+export const alwaysVisibleSelectorsMobile = {
+  footerWrapper: '.feds-footer-wrapper',
+  menuContent: '.feds-menu-content',
+  menuColumn: '.feds-menu-column',
+  menuSection: '.feds-menu-section',
+  ...socialLinksSelectors,
+  ...regionPickerSelectors,
+  ...legalSelectors,
+};
+
+// for small desktop and above
+export const alwaysVisibleSelectorsDesktop = {
+  footerWrapper: '.feds-footer-wrapper',
+  menuContent: '.feds-menu-content',
+  menuColumn: '.feds-menu-column',
+  menuSection: '.feds-menu-section',
+  ...featuredProductsSelectors,
+  ...socialLinksSelectors,
+  ...regionPickerSelectors,
+  ...legalSelectors,
+};
+
+export const allElementsVisible = (givenSelectors, parentEl) => {
+  const waitForElements = [];
+
+  const selectorsKeys = Object.keys(givenSelectors);
+  for (const selectorKey of selectorsKeys) {
+    const targetEl = parentEl.querySelector(givenSelectors[selectorKey]);
+    waitForElements.push(isElementVisible(targetEl));
+  }
+
+  return waitForElements.every((el) => el);
+};
+
+export const waitForFooterToDecorate = async () => {
+  const waitForElements = [];
+
+  const selectorsKeys = Object.keys(selectors);
+  for (const selectorKey of selectorsKeys) {
+    waitForElements.push(waitForElement(selectors[selectorKey]));
+  }
+  const elementsAreRendered = await Promise.all(waitForElements);
+  return !!elementsAreRendered;
+};
```

Comment:
You could shorten this:
```suggestion
export const waitForFooterToDecorate = () => Promise.all(
    Object
    .keys(selectors)
    .map(key => waitForElement(selectors[key]))
  );
```

And then just use
```js
// you don't need any checks, as the waitForElement method should timeout/reject and hence fail the Promise.all chain completely. If this runs through, you should be able to be confident all elements are now visible. (Or your checks in the test would fail anyway)
await waitForFooterToDecorate()
```